File: general/src/regex/lineEndingWeirdness.js
var control = 'abc';
var experimentWithLf = 'abc' + String.fromCharCode(10);
var experimentWithCr = 'abc' + String.fromCharCode(13);

var p1 = /^abc$/;
//var p2 = /\Aabc\z/; // not supported in JS

var result = {
    control : {
        'with ^$' : control.match(p1),
//        'with \A\z' : control.match(p2)
    },
    experimentWithLf : {
        'with ^$' : experimentWithLf.match(p1),
//        'with \A\z' : experimentWithLf.match(p2)
    },
    experimentWithCr : {
        'with ^$' : experimentWithCr.match(p1),
//        'with \A\z' : experimentWithCr.match(p2)
    }
};
console.dir(result);

File: general/src/regex/lineEndingWeirdness.rb
require 'pp'

control = 'abc'
experimentWithLf = control + 10.chr
experimentWithCr = control + 13.chr

p1 = "/^" + control + "$/"
p2 = "/\A" + control + "\z/"

result = {
    "control" => {
        "with ^$" => p1.match(control),
        "with \A\Z" => p2.match(control)
    },
    "experimentWithLf" => {
        "with ^$" => p1.match(experimentWithLf),
        "with \A\Z" => p2.match(experimentWithLf)
    },
    "experimentWithCr" => {
        "with ^$" => p1.match(experimentWithCr),
        "with \A\Z" => p2.match(experimentWithCr)
    }
}

pp result
